### DISCUSSION TITLE
Is it okay to have the output-dir set to a location that is outside the Quarto repo?

### DISCUSSION THREAD

**Discussion by neeldhara** [quarto-dev/quarto-cli#7133] [Q&A] [answered]
### Description

I have the following in my `quarto.yml`:

```qmd
project:
  type: website
  output-dir: "../../Dropbox/Notes/"
```

I hope this is fine? Seems to work out just fine - it puts the website in my Dropbox (I don't want to put my Quarto folder in dropbox because it's under version control, and DB has known issues syncing .git folders).

I want my local Quarto HTML output to be - (a) accessible when the local server is not running, and (b) accessible potentially in another machine (desktop or mobile) where I have Dropbox installed. I am thinking of saving local HTML file URLs in my bookmark system: I suppose even if I outgrow the Dropbox service, I can still have a folder named that way and the links will persist (even if I'll possibly lose the sync).

I just wanted to be sure there won't be any difficulties that I cannot anticipate with having the output-dir outside the repo.

**Comment by cscheid** [comment, marked as answer]
~~Although I'm happy that it works for you, I _think_ this is not intended behavior on our part: it's certainly inconsistent with other parts of quarto where we disallow project paths to resolve outside of the project folder. I believe that is a bug we need to fix (and sadly break your usage!)~~ 

~~I would, instead,~~ Consider a project post-render script to copy the files to the place you want to: https://quarto.org/docs/projects/scripts.html#pre-and-post-render

Edit: although the project script is still an option, this is actually intended behavior specifically for `output-dir`, in contrast to other project paths.

**Reply by neeldhara** [reply]
Ah, thank you for your prompt response here! I'm happy to setup a script, it's just that it creates a copy of the files, and I mostly open my files by search: so if I have two copies floating around on my system, it often ends up cluttering the search results, and I sometimes end up opening the wrong version and so on (although this being a view-only use case should not matter so much).

Long term I might worry about the double space usage but right now my output directory is < 1GB.

I suppose if the post-render script moved the files out, and a pre-render script put them back in, then I'd effectively have the same usage as I am enjoying right now? Would that also be not recommended? I can imagine render times go up because of a big folder being lugged around like that... once the output-dir does become large-ish.

Any chance you can leave this bug in as a feature? 🙈

**Reply by neeldhara** [reply]
> Any chance you can leave this bug in as a feature? 🙈

...or as an user-defined toggle? 👀

**Reply by cscheid** [reply]
> Any chance you can leave this bug in as a feature? 🙈

Good news! I talked to the other devs and it's actually intended behavior, as an exception for `output-dir` specifically, exactly for your use case. So we're not going to change this in the future.

**Reply by neeldhara** [reply]
> > Any chance you can leave this bug in as a feature? 🙈
> 
> Good news! I talked to the other devs and it's actually intended behavior, as an exception for `output-dir` specifically, exactly for your use case. So we're not going to change this in the future.

Yay! 💃  That's super helpful: very glad I can stick to this workflow, and I deeply appreciate your advice and confirmation here :)
